governance/program/src/instruction.rs

```diff
@@ -140,58 +130,64 @@ pub enum GovernanceInstruction {
         transfer_upgrade_authority: bool,
     },
 
-    /// Create Proposal account for Instructions that will be executed at various slots in the future
-    /// The instruction also grants Admin and Signatory token to the provided account
+    /// Creates Proposal account for Instructions that will be executed at various slots in the future
     ///
-    ///   0. `[writable]` Uninitialized Proposal account
-    ///   1. `[writable]` Initialized Governance account
-    ///   2. `[writable]` Initialized Signatory Mint account
-    ///   3. `[writable]` Initialized Admin Mint account
-    ///   4. `[writable]` Initialized Admin account for the issued admin token
-    ///   5. `[writable]` Initialized Signatory account for the issued signatory token
-    ///   6. '[]` Token program account
-    ///   7. `[]` Rent sysvar
+    ///   0. `[writable]` Proposal account. PDA seeds ['governance',account_governance, governing_token_mint, name]
```

Comment:
Ok, I've changed the PDA to use `proposal_index` based on `proposals_count` for `Governance` account.
I've been going back and forth on this because the index results in awkward api where you have to pass the index , which is based on the current Governance account `state`,  to instruction creator to get the correct PDA. However I agree with you it's lesser evil here.